### DISCUSSION TITLE
Custom colors on V2 produces an error

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [tailwindlabs/tailwindcss#2927] [Help]
So I'm in the middle of upgrading to V2 and am running into some errors with custom colors that are [defined as functions](https://github.com/adamwathan/tailwind-css-variable-text-opacity-demo). I keep getting this error message:

```bash
'colors.gray.400' was found but does not resolve to a string.

> 1 | @tailwind base;
    | ^
  2 | 
  3 | /* Write your own custom base styles here */
```

Because I'm using the method for colors that are functions (so transparency works), they shouldn't resolve to a string.

Here's the file for the new config, and it's being imported as a preset. https://github.com/Nick-Mazuk/ui-config/blob/master/tailwind.js

Any ideas? Is the process different for V2?

**Comment by MartinHughes-BPC** [comment]
Hi @Nick-Mazuk 

I think this is  bug - there's a PR that looks like what you've described: https://github.com/tailwindlabs/tailwindcss/pull/2919

Cheers,
Martin.

**Reply by Nick-Mazuk** [reply]
Certainly looks like the same issue.

**Reply by Nick-Mazuk** [reply]
@adamwathan Any ideas when the PR will be merged? Or is the process for adding custom colors different in V2.0? This is the only thing preventing me from using V2.0 right now.

P.S. Lots of thanks and appreciation for everything you've already done with Tailwind.

**Reply by jaredhill4** [reply]
@adamwathan, I'm having the same issue with V2, when I follow your example here:

https://github.com/adamwathan/tailwind-css-variable-text-opacity-demo

I'm getting:

```
'colors.gray.DEFAULT' was found but does not resolve to a string.
````

**Reply by jaredhill4** [reply]
@adamwathan, right now, this seems to only be an issue when using with the `theme()` function. `@apply` and using as a className in a component seem to be working just fine.

**Reply by Nick-Mazuk** [reply]
Looks like this issue was fixed as the PR was merged.

However, I'm now getting a similar issue with the Typeography plugin. https://github.com/tailwindlabs/tailwindcss-typography/issues/143

Any suggestions, @adamwathan ?

**Reply by jaredhill4** [reply]
@adamwathan - any updates here?
